doc/crypto/api/ops/ka.rst

```diff
@@ -114,6 +122,91 @@ Key agreement algorithms
 Standalone key agreement
 ------------------------
 
+.. function:: psa_key_agreement
+
+    .. summary::
+        Perform a key agreement and return the shared secret as a derivation key.
+
+    .. param:: psa_algorithm_t alg
+        The key agreement algorithm to compute: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_RAW_KEY_AGREEMENT(alg)` is true.
+    .. param:: psa_key_id_t private_key
+        Identifier of the private key to use.
+        It must permit the usage `PSA_KEY_USAGE_DERIVE`.
+    .. param:: const uint8_t * peer_key
+        Public key of the peer. The peer key must be in the same format that `psa_import_key()` accepts for the public key type corresponding to the type of ``private_key``. That is, this function performs the equivalent of :code:`psa_import_key(..., peer_key, peer_key_length)`, with key attributes indicating the public key type corresponding to the type of ``private_key``. For example, for ECC keys, this means that peer_key is interpreted as a point on the curve that the private key is on. The standard formats for public keys are documented in the documentation of `psa_export_public_key()`.
+    .. param:: size_t peer_key_length
+        Size of ``peer_key`` in bytes.
+
+    .. param:: const psa_key_attributes_t * attributes
```

Comment:
1. That comes from our general principle that inputs go before outputs.
2. When there are multiple keys involved, we put the attributes close to the corresponding key, to avoid confusion.
3. I think `psa_key_derivation_output_key()` should have had the attributes next to the key. Putting them first doesn't make sense given that everywhere else the operation goes first. (That is because it would be the self parameter in an object-oriented language.)
4. The position makes sense to me. As with (2), there are no key parameters between the attributes and the corresponding key.

Reply:
Note that In (1) the key attributes comes before the other _input parameters_, but not next to the output key parameter.

Otherwise I agree with this analysis. If we could redo, I would put the attributes input parameter immediately before the key output parameter in all these cases (this is the only output for all these functions). I.e. I would change (1) and (3).

So keep the position in this PR, and use this approach in the other forthcoming key-creation functions.

Reply:
Possibly the order (1) for [`psa_import_key`](https://arm-software.github.io/psa-api/crypto/1.1/api/keys/management.html#c.psa_import_key) is justified by the fact that the contents of the attributes affects how `data` is parsed, so it makes sense to put them before `data`.